### PR TITLE
Apollo gdt

### PR DESCRIPTION
**Pull request by einhornimmond**
## 🍰 Pullrequest
add function to list gdt transasctions

### CODE REVIEW

File: backend/src/graphql/models/GdtEntryList.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  summe: number
+}
+
+@ObjectType()
+export class GdtEntryList {
+  constructor(json: any) {
+    this.state = json.state
+    this.moreEntrysAsShown = json.moreEntrysAsShown
+    this.ownEntries = []
+    if (typeof json.ownEntries !== undefined) {
```
Comment (Mogge) [inline backend/src/graphql/models/GdtEntryList.ts:24]: Is this really necessary? Why not just:
`this.ownEntries = json.ownEntries ? json.ownEntries : []`

File: backend/src/graphql/models/GdtEntryList.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      for (const entry in json.ownEntries) {
+        // eslint-disable-next-line no-console
+        console.log(entry)
+        this.ownEntries.push(new GdtEntry(json.ownEntries[entry]))
+      }
+    }
+    this.gdtSumPerEmail = []
+    for (const email in json.gdtSumPerEmail) {
```
Comment (Mogge) [inline backend/src/graphql/models/GdtEntryList.ts:32]: `this.gdtSumPerEmail = json.gdtSumPerEmail`

**Review by Mogge — CHANGES_REQUESTED** [review]
Looks very good. Do my suggestions work? I could not test it.

**Comment by einhornimmond** [comment]
You can test it with this call:
```graphql
{
  listGDTEntries(
    email: "dariofrodo@gmx.de"
    firstPage: 1
    items: 5
    order: "DESC"
  ) {
    state
    ownEntries {
      amount
      gdt
      factor
    }
  }
}
```

File: backend/src/graphql/models/GdtEntryList.ts (hunk 1 above)
Comment (einhornimmond) [reply backend/src/graphql/models/GdtEntryList.ts:24]: When I replace my loop with your code I get this error:
"message": "Expected Iterable, but did not find one for field \"GdtEntryList.ownEntries\".",

File: backend/src/graphql/models/GdtEntryList.ts (hunk 2 above)
Comment (einhornimmond) [reply backend/src/graphql/models/GdtEntryList.ts:32]: I must first add more test data to test with this case

**Review by Mogge — CHANGES_REQUESTED** [review]

**Comment by Mogge** [comment]
This is how `json.ownEntries` looks like: 
```
{
  '1': {
    id: 1,
    amount: 99,
    date: '2020-01-01T13:47:12+00:00',
    email: 'dariofrodo@gmx.de',
    comment: '',
    coupon_code: null,
    gdt_entry_type_id: 1,
    factor: '18.0000',
    amount2: 0,
    factor2: '1.0000',
    gdt_modificator_entry: null,
    gdt: 1782
  },
  '6': {
    id: 6,
    amount: 1782,
    date: '2021-07-30T13:21:46+00:00',
    email: 'dariofrodo@gmx.de',
    comment: 'Event 20-20',
    coupon_code: null,
    gdt_entry_type_id: 7,
    factor: '0.2000',
    amount2: 0,
    factor2: '1.0000',
    gdt: 356.40000000000003
  }
}
```

I would expect:
```
[
  {
    id: 1,
    amount: 99,
    date: '2020-01-01T13:47:12+00:00',
    email: 'dariofrodo@gmx.de',
    comment: '',
    coupon_code: null,
    gdt_entry_type_id: 1,
    factor: '18.0000',
    amount2: 0,
    factor2: '1.0000',
    gdt_modificator_entry: null,
    gdt: 1782
  },
  {
    id: 6,
    amount: 1782,
    date: '2021-07-30T13:21:46+00:00',
    email: 'dariofrodo@gmx.de',
    comment: 'Event 20-20',
    coupon_code: null,
    gdt_entry_type_id: 7,
    factor: '0.2000',
    amount2: 0,
    factor2: '1.0000',
    gdt: 356.40000000000003
  }
]
```
and my suggestion works

**Comment by Mogge** [comment]
In the log of your solution I see:
```
1
6
```

**Comment by ulfgebhardt** [comment]
@einhornimmond @Mogge Can you guys make a pairing over this and get this merged since the backend is now merged in?

**Review by Mogge — APPROVED** [review]
Topp!

**Pull request merged**